sys-utils/ipcutils.c

```diff
@@ -227,6 +251,75 @@ void ipc_shm_free_info(struct shm_data *shmds)
 	}
 }
 
+int posix_ipc_shm_get_info(const char *name, struct posix_shm_data **shmds)
+{
+	DIR *d;
+	int i = 0;
+	struct posix_shm_data *p;
+	struct dirent *de;
+
+	p = *shmds = xcalloc(1, sizeof(struct posix_shm_data));
+	p->next = NULL;
+
+	d = opendir(_PATH_DEV_SHM);
+	if (!d)
+		err(EXIT_FAILURE, _("cannot open %s"), _PATH_DEV_SHM);
+
```

Comment:
The code should be ready for situations when the directory is not available. My suggestion:
```
d = opendir(_PATH_DEV_SHM);
if (!d)
   return -1;

p = *shmds = xcalloc(1, sizeof(struct posix_shm_data));
...
```
It means return -1 (or -errno) rather than err(); alloc when sure the directory is accessible.